src/fragments/lib/geo/js/geofences.mdx

```diff
@@ -52,3 +52,328 @@ initializeMap();
+
+#### API
+
+```javascript
+Geo.saveGeofences(geofences, options);
+```
+
+#### Parameters
+
+- `geofences` - can be a single geofence object, or an array of geofence objects to save to a collection.
+
+Geofence objects must have the following properties:
+
+- `geofenceId` - a opaque and unique identifier for the geofence.
+- `geometry` - a geometry object that defines the geofence.
+  - `polygon` - an array of arrays with [Longitude, Latitude] coordinates.
+
+<Callout>
+
+NOTE: Polygon arrays have a few requirements:
+
+- must have at least 4 vertices (i.e. 4 coordinate points)
+- the first and last point must be the same in order to complete the polygonal loop
+- vertices must be in counter-clockwise order
+
+</Callout>
+
+```json
+{
+  "geofenceId": "string",
+  "geometry": {
+    "polygon": [
+      [
+        [longitude, latitude],
+        [longitude, latitude],
+        [longitude, latitude],
+        [longitude, latitude]
+      ]
+    ]
+  }
+}
+```
+
+- `options` - optional options object for saving geofences
+  - `collectionName` - the name of the collection to save geofences to.
+    - Defaults to the default collection listed in your `aws-exports.js` file after provisioning a geofence collection resource.
+
+```js
+{
+  collectionName: "myCollection";
+}
+```
+
+#### Return
+
+The return from `saveGeofences` is a Promise that resolves to an object with both successes and errors for geofences that were successfully created or not.
+
+Each success object has the following properties:
+
+- `geofenceId` - the geofenceId of the geofence that was saved.
+- `createTime` - the time the geofence was created.
+- `updateTime` - the time the geofence was last updated.
+
+Each error object has the following properties:
+
+- `geofenceId` - the geofenceId of the geofence that failed to be saved.
+- `error` - an error object
+  - `code` - the error code
+  - `message` - the error message
+
+```js
+// Promise<SaveGeofenceResults>
+{
+  successes: [
+    {
+      geofenceId: "string",
+      createTime: Date,
+      updateTime: Date,
+    },
+  ],
+  errors: [
+    {
+      error: {
+        code: "string",
+        message: "string",
+      },
+      geofenceId: "string",
+    },
+  ],
+};
+```
+
+#### Example
+
+```js
+let saveGeofenceResults;
+try {
+  saveGeofenceResults = await Geo.createGeofence({
+    geofenceId: "my-geofence",
+    geometry: {
+      polygon: [
+        [-123.14695358276366, 49.290090146520434],
+        [-123.1358814239502, 49.294960279811974],
+        [-123.15021514892577, 49.29300108863353],
+        [-123.14909934997559, 49.29132171993048],
+        [-123.14695358276366, 49.290090146520434],
+      ],
+    },
+  });
+} catch (error) {
+  // error handling from logic and validation issues within `createGeofences`
+  throw error;
+}
+
+if (saveGeofenceResults.errors.length > 0) {
+  // error handling that are from the underlying API calls
+  throw new Error();
+}
```

Comment:
Whether to throw here or not is an application level logic. We should just print out the number of successes and errors to show how to access the response object.